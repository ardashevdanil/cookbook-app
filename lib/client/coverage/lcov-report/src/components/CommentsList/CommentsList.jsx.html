<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/CommentsList/CommentsList.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/CommentsList</a> CommentsList.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React from 'react';
import './CommentsList.css';
&nbsp;
import Comment from '../Comment/Comment';
import CommentInput from '../CommentInput/CommentInput';
&nbsp;
class CommentsList extends React.Component {
  componentDidMount() {
    this.props.fetchComments();
  }
&nbsp;
  render() {
    const {
      comments,
      fetchComments,
      recipe,
      user,
    } = { ...this.props };
    const commentsComponents = comments.map(comment =&gt; (
      &lt;Comment
        avatar={comment.avatar}
        date={comment.date}
        likes={comment.likes}
        key={comment.id}
        text={comment.text}
        userName={comment.userName}
      /&gt;
    ));
&nbsp;
    <span class="missing-if-branch" title="if path not taken" >I</span>if (!comments.length) {
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;CommentInput
          fetchComments={fetchComments}
          recipe={recipe}
          user={user}
        /&gt;
      );
    }
&nbsp;
    return (
      &lt;div&gt;
        &lt;div className="CommentsList"&gt;
          &lt;div className="CommentsList__heading"&gt;
            Comments
          &lt;/div&gt;
          &lt;div className="CommentsList__list"&gt;
            {commentsComponents}
          &lt;/div&gt;
        &lt;/div&gt;
        &lt;CommentInput
          fetchComments={fetchComments}
          recipe={recipe}
          user={user}
        /&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
export default CommentsList;
&nbsp;
CommentsList.defaultProps = {
  comments: [],
};
&nbsp;
CommentsList.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.object),
  fetchComments: PropTypes.func.isRequired,
  recipe: PropTypes.number.isRequired,
  user: PropTypes.string.isRequired,
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 20:41:16 GMT+0300 (MSK)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
