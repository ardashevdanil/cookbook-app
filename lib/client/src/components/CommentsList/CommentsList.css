.CommentsList {
  box-sizing: border-box;
  margin: 45px 0;
  padding: 20px 40px 30px 40px;
  background-color: white;
  box-shadow: 0 0 13px 0 rgba(11, 12, 12, 0.15);
}

.CommentsList__heading {
  font-size: 24px;
  padding-bottom: 9px;
  border-bottom: 2px solid #ddd;
}

@media only screen and (max-width: 1120px) {
  .CommentsList {
    margin: 0;
    padding: 11.1vw 6.9vw;
    box-shadow: none;
  }

  .CommentsList__heading {
    font-size: 7vw;
    padding-bottom: 2.8vw;
    border-width: 0.7vw;
  }
}
