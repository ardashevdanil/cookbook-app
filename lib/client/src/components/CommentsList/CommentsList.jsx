import PropTypes from 'prop-types';
import React from 'react';
import './CommentsList.css';

import Comment from '../Comment/Comment';
import CommentInput from '../CommentInput/CommentInput';

class CommentsList extends React.Component {
  componentDidMount() {
    this.props.fetchComments();
  }

  render() {
    const {
      comments,
      fetchComments,
      recipe,
      user,
    } = { ...this.props };
    const commentsComponents = comments.map(comment => (
      <Comment
        avatar={comment.avatar}
        date={comment.date}
        likes={comment.likes}
        key={comment.id}
        text={comment.text}
        userName={comment.userName}
      />
    ));

    if (!comments.length) {
      return (
        <CommentInput
          fetchComments={fetchComments}
          recipe={recipe}
          user={user}
        />
      );
    }

    return (
      <div>
        <div className="CommentsList">
          <div className="CommentsList__heading">
            Comments
          </div>
          <div className="CommentsList__list">
            {commentsComponents}
          </div>
        </div>
        <CommentInput
          fetchComments={fetchComments}
          recipe={recipe}
          user={user}
        />
      </div>
    );
  }
}

export default CommentsList;

CommentsList.defaultProps = {
  comments: [],
};

CommentsList.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.object),
  fetchComments: PropTypes.func.isRequired,
  recipe: PropTypes.number.isRequired,
  user: PropTypes.string.isRequired,
};
