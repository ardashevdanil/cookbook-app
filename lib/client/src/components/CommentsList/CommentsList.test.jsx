import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import CommentsList from './CommentsList';

Enzyme.configure({ adapter: new Adapter() });

describe('test CommentsList component', () => {
  const component = Enzyme.shallow(
    <CommentsList
      comments={[{}]}
      fetchComments={() => null}
      recipe={1}
      user="test"
    />,
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
