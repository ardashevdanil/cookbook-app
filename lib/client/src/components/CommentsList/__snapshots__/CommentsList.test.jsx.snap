// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CommentsList component renders correctly 1`] = `
"<div>
  <div className=\\"CommentsList\\">
    <div className=\\"CommentsList__heading\\">
      Comments
    </div>
    <div className=\\"CommentsList__list\\">
      <Comment avatar={[undefined]} date={[undefined]} likes={[undefined]} text={[undefined]} userName={[undefined]} />
    </div>
  </div>
  <CommentInput fetchComments={[Function: fetchComments]} recipe={1} user=\\"test\\" />
</div>"
`;
